packages/connect-web/src/popup/index.ts

```diff
@@ -281,12 +281,15 @@ export class PopupManager extends EventEmitter {
             // send ConnectSettings to popup
             // note this settings and iframe.ConnectSettings could be different (especially: origin, popup, webusb, debug)
             // now popup is able to load assets
-        } else if (data.type === POPUP.CANCEL_POPUP_REQUEST || data.type === UI.CLOSE_UI_WINDOW) {
-            this.close();
+        } else if (
+            data.type === POPUP.CANCEL_POPUP_REQUEST
+            //  || data.type === UI.CLOSE_UI_WINDOW // Ignoring because it is always sent with CANCEL_POPUP_REQUEST
```

Comment:
seems so. as long as there is popupPromise at the moment of calling closePopup()
```
const closePopup = (success: boolean) => {
    if (_popupPromise) {
        postMessage(createPopupMessage(POPUP.CANCEL_POPUP_REQUEST, { success }));
    }
    postMessage(createUiMessage(UI.CLOSE_UI_WINDOW));
};
```
this is just 🦆 method. In general I am all for simplifying stuff if possible

Reply:
I left it as it was.